src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs

```diff
@@ -25,6 +23,7 @@ internal class AppiumCommandExecutor : ICommandExecutor
         private ICommandExecutor RealExecutor;
         private bool isDisposed;
         private const string IdempotencyHeader = "X-Idempotency-Key";
+        private const int MaxRetryAttempts = 3;
```

Comment:
do you know why command failures happen? We never retry session creation in other clients, especially implicitly
In java client , for example, the only situation where a failure could happen is the client read timeout (e.g. the session creation takes more than the client http timeout). Setting the read timeout to 10 minutes helped to solve the issue.

Reply:
Not sure why it happen.
So I'll remove the retry mechanism, anyway I didn't liked that approach.
I'll try playing with the read timeout if you suggest this solution. Can you provide a link to the place this magic happen on the Java client maybe?

Reply:
@mykola-mokhnach, to your question. 
here is an example on the first try I get noSuchDriver, but if I try again I receive a success status
first try:
![image](https://github.com/appium/dotnet-client/assets/59066376/fe53f236-cb61-4d1b-8d15-834d79d32174)
2nd try:
![image](https://github.com/appium/dotnet-client/assets/59066376/662dcc29-91dd-4cec-b215-f738bffb842f)

Reply:
do you also have the server log?

Reply:
sure. But I didn't see any logs related to `noSuchDriver` on the server.
https://gist.github.com/Dor-bl/a51a074d8e1aa4d8c02d242adb8067c8

Reply:
I am not sure why it requests for log types on session creation, but NoSuchDriver error usually happens if the session dies because of some internal error/crash, so then all the following commands to the session identifier returned by NewSession API are not valid anymore.

Reply:
yes, I'm aware of this. but still, I get this error right after a new session creation and on the 2nd try it returns success

Reply:
@mykola-mokhnach, when you said to set client read timeout to 10 minutes, you mean here?
![image](https://github.com/appium/dotnet-client/assets/59066376/dd738b18-b2b0-44e7-8fbc-59921569a4db)

Reply:
same idea, no?
anyway, do you think it's more elegant to catch this case and retry again? only if noSuchdriver issue appear at first try

Reply:
to me it feels like a workaround rather than a proper fix. 
I am not going to completely deny this PR, because I don't know much about the particular details of dotnet implementation and I trust you as the project maintainer.

Maybe you have anything to add on this @KazuCocoa ?

Reply:
So for now I'll leave the noSuchDriver issue. I made other required changes for the greater good. such as XML summaries and some code improvements

Reply:
Sounds good in current pr.
I cannot recall properly, but probably I saw similar "no such driver" error message before. At that time was... maybe uia2/xcuitest's uia2 server/wda got failed or something in a new session request. Probably such a thing occurred...

but in this case, passing such an error to the caller (so test code) by default is probably good. Retrial could be an option though.

In my past experience, webdriverio had such retrial by default. it confused users so I needed to suggest disabling them first to make the failure reason clear. Thus maybe do not retry by default is basically better, but option could be. (this is my personal opinion)

Reply:
Maybe I can add an option to to appium options class, that if the user set it's value to true it will activate retry for this case. this field will of course will be false by default.